openlibrary/plugins/openlibrary/js/readmore.js

```diff
@@ -44,18 +48,23 @@ export class ReadMoreComponent {
     expand() {
         this.$container.classList.add('read-more--expanded');
         this.$content.style.maxHeight = `${this.fullHeight}px`;
+        this.$readMoreButton.style.display = 'none';
+        this.$readLessButton.style.display = 'block';
     }
 
     collapse() {
         this.$container.classList.remove('read-more--expanded');
         this.$content.style.maxHeight = `${this.collapsedHeight}px`;
+        this.$readMoreButton.style.display = 'block';
+        this.$readLessButton.style.display = 'none';
     }
 
     reset() {
+        if (!this.$content || !this.$readMoreButton) return;
         this.fullHeight = this.$content.scrollHeight;
         // Fudge factor to account for non-significant read/more
         // (e.g missing a bit of padding)
-        if (this.$content.scrollHeight <= (this.collapsedHeight + 1)) {
+        if (this.$content.scrollHeight <= (this.collapsedHeight + this.$readMoreButton.offsetHeight + 1)) {
```

Comment:
This should be the only change that is necessary here. Please revert the other changes since they are unnecessary.